src/main/java/reactor/kafka/receiver/internals/ConsumerEventLoop.java

```diff
@@ -53,80 +54,63 @@
     // TODO make it final
     org.apache.kafka.clients.consumer.Consumer<K, V> consumer;
 
-    CoreSubscriber<? super ConsumerRecords<K, V>> actual;
+    final FluxSink<ConsumerRecords<K, V>> sink;
 
     final AtomicBoolean awaitingTransaction;
 
-    ConsumerFlux(
+    ConsumerEventLoop(
         AckMode ackMode,
+        AtmostOnceOffsets atmostOnceOffsets,
         ReceiverOptions<K, V> receiverOptions,
-        Scheduler eventScheduler, org.apache.kafka.clients.consumer.Consumer<K, V> consumer,
+        Scheduler eventScheduler,
+        org.apache.kafka.clients.consumer.Consumer<K, V> consumer,
         Predicate<Throwable> isRetriableException,
+        FluxSink<ConsumerRecords<K, V>> sink,
         AtomicBoolean awaitingTransaction
     ) {
         this.ackMode = ackMode;
+        this.atmostOnceOffsets = atmostOnceOffsets;
         this.receiverOptions = receiverOptions;
         this.eventScheduler = eventScheduler;
         this.consumer = consumer;
         this.isRetriableException = isRetriableException;
+        this.sink = sink;
         this.awaitingTransaction = awaitingTransaction;
 
         pollEvent = new PollEvent();
-    }
-
-    @Override
-    public void subscribe(CoreSubscriber<? super ConsumerRecords<K, V>> actual) {
-        log.debug("start");
 
-        if (!isActive.compareAndSet(false, true)) {
-            Operators.error(actual, new IllegalStateException("Multiple subscribers are not supported for KafkaReceiver flux"));
-            return;
+        eventScheduler.schedule(new SubscribeEvent());
+
+        Duration commitInterval = receiverOptions.commitInterval();
+        if (!commitInterval.isZero()) {
+            switch (ackMode) {
+                case AUTO_ACK:
+                case MANUAL_ACK:
+                    eventScheduler.schedulePeriodically(
```

Comment:
As a separate improvement I would consider grabbing the returned `Disposable` to allow for `stop()` method to dispose it, maybe?

Reply:
Isn't it handled by `Scheduler#dispose`?

Reply:
I missed that in the `stop()` method, somehow. 👍